broker-core/src/test/java/io/zeebe/broker/workflow/message/MessageCatchElementTest.java

```diff
@@ -109,212 +125,231 @@ public void init() {
-            "receive-message", WorkflowInstanceIntent.ELEMENT_ACTIVATED);
-
-    // when
-    final DirectBuffer messagePayload = asMsgPack("foo", "bar");
-    testClient.publishMessage("order canceled", "order-123", messagePayload);
-
-    // then
-    final Record<WorkflowInstanceSubscriptionRecordValue> subscription =
-        testClient
-            .receiveWorkflowInstanceSubscriptions()
-            .withIntent(WorkflowInstanceSubscriptionIntent.CORRELATED)
-            .getFirst();
-
-    assertThat(subscription.getMetadata().getValueType())
-        .isEqualTo(ValueType.WORKFLOW_INSTANCE_SUBSCRIPTION);
-    assertThat(subscription.getMetadata().getRecordType()).isEqualTo(RecordType.EVENT);
-
-    assertWorkflowSubscription(
-        workflowInstanceKey, "{\"foo\":\"bar\"}", catchEventEntered, subscription);
-  }
-
-  @Test
-  public void shouldCorrelateMessageSubscription() {
-    // given
-    final long workflowInstanceKey =
-        testClient.createWorkflowInstance(PROCESS_ID, asMsgPack("orderId", "order-123"));
-
-    final Record<WorkflowInstanceRecordValue> catchEventEntered =
-        testClient.receiveElementInState(
-            "receive-message", WorkflowInstanceIntent.ELEMENT_ACTIVATED);
-
-    // when
-    testClient.publishMessage("order canceled", "order-123", asMsgPack("foo", "bar"));
-
-    // then
-    final Record<MessageSubscriptionRecordValue> subscription =
-        testClient
-            .receiveMessageSubscriptions()
-            .withIntent(MessageSubscriptionIntent.CORRELATED)
-            .getFirst();
-
-    assertThat(subscription.getMetadata().getValueType()).isEqualTo(ValueType.MESSAGE_SUBSCRIPTION);
-    assertThat(subscription.getMetadata().getRecordType()).isEqualTo(RecordType.EVENT);
-
-    assertMessageSubscription(workflowInstanceKey, catchEventEntered, subscription);
-  }
-
-  @Test
-  public void shouldCloseMessageSubscription() {
-
-    final long workflowInstanceKey =
-        testClient.createWorkflowInstance(PROCESS_ID, asMsgPack("orderId", "order-123"));
-
-    final Record<WorkflowInstanceRecordValue> catchEventEntered =
-        testClient.receiveElementInState(
-            "receive-message", WorkflowInstanceIntent.ELEMENT_ACTIVATED);
-
-    testClient.cancelWorkflowInstance(workflowInstanceKey);
-
-    final Record<MessageSubscriptionRecordValue> messageSubscription =
-        RecordingExporter.messageSubscriptionRecords(MessageSubscriptionIntent.CLOSED).getFirst();
-
-    assertThat(messageSubscription.getMetadata().getRecordType()).isEqualTo(RecordType.EVENT);
-
-    Assertions.assertThat(messageSubscription.getValue())
-        .hasWorkflowInstanceKey(workflowInstanceKey)
-        .hasElementInstanceKey(catchEventEntered.getKey())
-        .hasMessageName("order canceled")
-        .hasCorrelationKey("");
+  /** Test cases which only apply to catch elements in normal flow */
+  @RunWith(Parameterized.class)
+  public static class NormalFlowTestCases extends MessageCatchElementTest {
```

Comment:
I think you don't need to embed test classes. We can extract the lifecycle test and delete the other one.

Reply:
The lifecycle test doesn't really apply to the boundary event case, unfortunately (and that's already tested elsewhere), and I couldn't find anything about skipping a test for a single set of parameters. Is there something like that? :thinking: